common/app-starters-task-launch-request-common/src/main/java/org/springframework/cloud/stream/app/tasklaunchrequest/DataflowTaskLaunchRequestProperties.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.tasklaunchrequest;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.validation.annotation.Validated;
+
+/**
+ * Base Properties to create a {@link org.springframework.cloud.task.launcher.TaskLaunchRequest}.
+ *
+ * @author Chris Schaefer
+ * @author David Turanski
+ */
+@Validated
+@ConfigurationProperties("task.launch.request")
+public class DataflowTaskLaunchRequestProperties {
+
+	/**
+	 * Comma separated list of optional args in key=value format.
+	 */
+	private List<String> args = new ArrayList<>();
+
+	/**
+	 * Comma delimited list of deployment properties to be applied to the
+	 * TaskLaunchRequest.
+	 */
+	private String deploymentProperties;
```

Comment:
What is the difference in the interpretation for the `args` and this `deploymentProperties`?
Why they are both based on the *Comma delimited list*, but they have different types?

Reply:
Command line args may or may not be k=v.  Properties must be and eventually transformed to a map.